[Anthony Ho] iP

anthonyhoth opened the pull request:
## Jukebox
> “When the going gets tough, the tough get going." 
> – Joe Kennedy

**Jukebox** frees your mind of having to remember things you need to do. 

It's,
- funny
- apologetic
- ~FAST~ SUPER FAST
- All you need to do is,

download it from [here](https://github.com/anthonyhoth/ip/releases).
double-click it.
add your tasks.
let it manage your tasks for you 😉

And it is **FREE**!

**Features**:
-  Managing tasks
-  Managing deadlines (_coming soon_)
-  Reminders (_coming soon_)

Commit:
Add Gradle support

anttthho committed:
Level 1 done

anttthho committed:
Completed Level 2

anttthho committed:
Completed Level 3

anttthho committed:
Completed TextUiTesting

anttthho committed:
Completed Level-5

anttthho committed:
Completed Level 6

anttthho committed:
Completed Level 6

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+import java.util.ArrayList;
+
 public class Duke {
+    // Initialise the task list
```
Would it be better if you phrase your comment as a noun eg. `The task list`? Because I think verbs should be reserved for methods.

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+
+
+        /**
+         * Prints out the task list iteratively.
```
Should there be a new line separating the JavaDoc description and params?

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+         * Prints out the task list iteratively.
+         * @param taskList Input task list
+         */
+        private static void printList (ArrayList < Task > taskList) {
```
Are the whitespaces in `ArrayList < Task > ` necessary?

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+
+        /**
+         * Prints out the task list iteratively.
+         * @param taskList Input task list
```
Should there be a period at the end of your sentences? (i.e.`@param taskList Input task list.`)

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+         * @param command Input command word
+         * @param inputWordsSplit String array of the command word with the task number
+         */
+        private static void markUnmarkTask (CommandWord command, String[] inputWordsSplit){
```
I think there should be a whitespace before `{`. Also, can the method be better named (eg. `unmarkTask`)?

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+        /**
+         * Enum for the command word
+         */
+        private enum CommandWord {
```
Nice use of enums 😄

ngshijun replied:
Good use of enum 👍

njxue commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+             */
+            private static CommandWord getCommand(String input) throws IllegalArgumentException {
+                for (int i = 0; i < CommandWord.values().length; i++) {
+                    CommandWord temp = CommandWord.values()[i];
```
Could the `temp` variable be better renamed (e.g. `commandWord`)?

njxue reviewed (commented):
There are some minor inconsistences in the use of JavaDocs:
- Missing blank line separating the description and params
- Missing period at the end of each sentence

Would also be nice if you could write JavaDocs for classes.

Other than that, LGTM! 💯

ngshijun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,210 @@
+                        return temp;
+                    }
+                }
+                throw new IllegalArgumentException("There is no such command called " + input + "!\n" + "Please try again :)\n");
```
instead of throwing an `IllegalArgumentExcrption`, perhaps you can wrap the exception into a new `DukeException`.

ngshijun reviewed (commented):
Looks good to me so far. Try to catch up with the progress.

anttthho committed:
Commit Duke and Event for Level 6

anttthho committed:
Completed Level-7

anttthho committed:
Completed Level-8

anttthho committed:
Merging branch-Level-8

anttthho committed:
Refactored Duke into Classes

anttthho committed:
Divide Duke Classes into Packages

anttthho committed:
Merge 'add-gradle-support' Branch

anttthho committed:
Add JUnit Tests

anttthho committed:
Add JavaDoc Comments

anttthho committed:
Fix JUnit Tests

anttthho committed:
Add Find Command

anttthho committed:
Merge branch 'branch-A-CodingStandard'

anttthho committed:
Merge branch 'branch-Level-9'

anttthho committed:
Integrated Checkstyle and fixed Checkstyle errors

anttthho committed:
Added GUI Functionality

anttthho committed:
Added assertions feature to document important assumptions

Fixed build.gradle for gradlew shadowjar

Fixed Checkstyle errors

anttthho committed:
Some methods in TaskList do not adhere to good standards of code
quality.

This will lead to poor readability of code and hence, understandability.
In any non-trivial software project, code needs to be read, understood,
and modified by other developers later on.

Further extracting out long methods into many smaller methods will help
the reader to better understand our code.

Separating into smaller helper codes, the reader will be able to read
the code better at a glance, instead of tracing through one long method.

anthonyhoth committed:
Merge pull request #3 from anthonyhoth/branch-A-Assertions

Added assertions feature to document important assumptions & Fixed errors

anthonyhoth committed:
Merge pull request #4 from anthonyhoth/branch-A-CodeQuality

Improve Code Quality of TaskList Class

anttthho committed:
Adding basic note management functionality

anttthho committed:
Completed Note Management Functionality

Added "note" command to add notes to any Todo, Deadline or Event.

Edited Storage to save/load accordingly with new Notes.
Edited Parser to parse input of Note command.
Edited TaskList to print task list with new Notes.

anthonyhoth committed:
Merge pull request #5 from anthonyhoth/branch-BCD-Extension

Completed Note Management Functionality

anttthho committed:
Added GUI tweaks

anttthho committed:
Added a unique cat personality to Duke

anttthho committed:
Added screenshot of product as Ui.png

anttthho committed:
Updated README.md with Toothless User Guide

Fix phrases used by Duke in Ui and Main
